### ISSUE TITLE
Separate DPI scaling for a popped out window

### ISSUE THREAD

**Issue by btglau** [spyder-ide/spyder#5325]
## Description of your problem

When I launch spyder, I like to have the code window on my laptop screen, and the rest of the windows such as the console, history, file browser, etc in a big down that is on a larger screen. The big monitor is at 100% dpi scaling, while my laptop screen is at 125%. However, the text on the 125% scaling monitor remains small.

**What steps will reproduce the problem?**

1. Pop out a window from the spyder IDE
2. Have two windows, each with different DPI scaling.
3. When moving windows back and forth, each window adopts the DPI scaling of monitor that the 'main' window is in.

**What is the expected output? What do you see instead?**
Each window adopts a DPI scaling based upon the monitor it is on.

**Please provide any additional information below**

MATLAB will open a second editor window in Windows 10 - i.e., if I hover my mouse over the matlab icon on the taskbar, two windows will show up, whereas in spyder, only a single window shows up. That suggests to me that the spyder gui is treating the popped out code window as part of the main window, even though I would like them to behave separately.

## Versions and main components

* Spyder Version: 3.2.3
* Python Version: 3.6.2 64 bit
* Qt Version: 5.6.2
* PyQt Version: 5.6
* Operating system: Windows 10

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:

IPython >=4.0    :  6.1.0 (OK)
cython >=0.21    :  0.26 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
numpy >=1.7      :  1.13.1 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.6.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.7.2 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
sympy >=0.7.3    :  1.1.1 (OK)

**Comment by jnsebgosselin** [comment]
@btglau  Thanks for reporting this issue.

@ccordoba12 Maybe this is going to be fixed as part of PR #3824?

**Comment by ccordoba12** [comment]
Yes, I think this is going to be fixed in PR #3284. @btglau, that work is for Spyder 4 though, to be released next year.

@dalthviz, can you test that that's the case?

**Comment by btglau** [comment]
@ccordoba12 That's great news! Being able to arrange spyder to look like matlab was a great way to get me started with python. Thanks for paying attention to small details like this.

**Comment by dalthviz** [comment]
@ccordoba12 I don't know how to test the DPI scaling per monitor since I'm working with Windows 8.1 (which I think only uses one level of scaling for all the displays or sets it automatically). However, with #3284 you get this behavior:

![window](https://user-images.githubusercontent.com/16781833/30842997-eb571d32-a24b-11e7-85d2-5166c7eddc62.gif)

@btglau could you try the PR #3824 and see the behavior in your monitors?

Also, thanks @jnsebgosselin for pointing the possibility of the relation of that PR with this issue :+1:

**Comment by btglau** [comment]
@dalthviz 

Terribly sorry for the slow reply - should have time to test later this week. I am new to github - any quick instruction on trying out this beta?

(I am using spyder via anaconda)

**Comment by dalthviz** [comment]
Hi @btglau for trying this you can: 

Using `git` (you can download it [here](https://git-scm.com/download/win)) in a cmd and without any other Spyder instance running:
```cmd
git clone https://github.com/dalthviz/spyder.git
cd spyder
git checkout fixes_issue_3970_2
python bootstrap.py
```
Another option could be to replace the files that were changed in the PR #3824 in your local installation of Spyder (the files are in the [branch fixes_issue_3790_2](https://github.com/dalthviz/spyder/tree/fixes_issue_3790_2) and your local installation of Spyder in your Anaconda installation at `...\Anaconda\Lib\site-packages\spyder\`)

Thanks for the testing! :)

**Comment by groundhogxc** [comment]
Hi, I tested the branch 3790_2, but I do not have the impression that the issue is fixed (although I am not quite sure I am looking for the right thing).

If I move a *docked* editor from Screen1 to Screen 2, the scaling seems to switch as soon as the main window (center, or mouse) flips to the new screen (and there are some rendering errors as a result). If I *undock* the editor window, it does not switch if I move it alone. However, it will change the scaling when I move the main window also to the second screen. (Except not immediately in that case, I need to click into the undocked editor first. Also, there are rendering errors when I do that, like in the case above).

On the rendering errors (However I suspect that is a separate issue from this one): In those cases the editor mostly uses wrong fonts/font sizes so the text looks funny. Also the line spacing does not match the font. 
Most of these text rendering errors are fixed when I switch to another editor tab and back (but not all of them) 

Dependencies: (Windows 10)
## Version and main components

* Spyder Version: 4.0.0.dev0 43a55ad2e
* Python Version: 3.6.2
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.5.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
numpy >=1.7      :  1.13.1 (OK)
sphinx >=0.6.6   :  1.6.2 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
cython >=0.21    :  0.26 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
IPython >=4.0    :  6.1.0 (OK)
pylint >=0.25    :  1.6.4 (OK)

```

**Comment by dalthviz** [comment]
Thanks @groundhogxc for the testing :+1:
